psrecord/main.py

```diff
@@ -45,19 +47,20 @@ def get_memory(process):
 
 
 def all_children(pr):
-    processes = []
-    children = []
+    global children
+
     try:
-        children = pr.children()
+        children_of_pr = pr.children()
     except AttributeError:
-        children = pr.get_children()
+        children_of_pr = pr.get_children()
```

Comment:
by the way `pr.get_children` is deprecated in `psutil`, you might consider drop support for legacy version
refs https://github.com/pythonprofilers/memory_profiler/issues/101

Reply:
Done